frontend/magic_links/routes.py

```diff
@@ -12,9 +19,53 @@
 
 @magic_links_bp.route("/invalid")
 def invalid():
-    new_magic_link_url = env.config.get("NEW_MAGIC_LINK_URL")
 
     return (
-        render_template("invalid.html", new_magic_link_url=new_magic_link_url),
+        render_template(
+            "invalid.html", new_magic_link_url=url_for("magic_links_bp.new")
+        ),
         403,
     )
+
+
+@magic_links_bp.route("/new", methods=["GET", "POST"])
+def new():
+    """
+    Returns a page containing a single question requesting the
+    users email address.
+    """
+    fund_id = request.args.get("fund_id")
+    round_id = request.args.get("round_id")
+
+    form_data = request.data
+    if request.method == "GET":
+        form_data = request.args
+
+    form = EmailForm(data=form_data)
+
+    if form.validate_on_submit():
+        try:
+            AccountMethods.get_magic_link(
+                email=form.data.get("email"),
+                fund_id=fund_id,
+                round_id=round_id,
+            )
+            return redirect(url_for("magic_links_bp.check_email"))
+        except MagicLinkError as e:
+            form.email.errors.append(str(e.message))
+        except NotificationError as e:
+            form.email.errors.append(str(e.message))
+        except AccountError as e:
+            form.email.errors.append(str(e.message))
```

Comment:
💯 Really nice to see that each service is starting to get its own error on failure.

Reply:
On it